link.go

```diff
@@ -95,7 +95,7 @@ func newLink(s *Session, r *Receiver, opts []LinkOption) (*link, error) {
 }
 
 // attachLink is used by Receiver and Sender to create new links
-func attachLink(s *Session, r *Receiver, opts []LinkOption) (*link, error) {
+func attachLink(ctx context.Context, s *Session, r *Receiver, opts []LinkOption) (*link, error) {
```

Comment:
It's good to add a context here. Do you know if we have to unroll anything on our way out if we cancel though?

Reply:
I don't believe so.  I say this because there were already other error cases that would return early and the link mux isn't started until the very end.